### ISSUE TITLE
targets

### ISSUE THREAD

**Issue by 1206927506** [jacobgil/pytorch-grad-cam#436]
There are a number of different classes in the model_targets file, But I do not understand ClassifierOutputTarget, ClassifierOutputSoftmaxTarget, SoftmaxOutputTarget, RawScoresOutputTarget difference and application scenarios. Specifying how much to view class_num is crucial for generating the heat map, and I hope you can help me.

**Comment by 1206927506** [comment]
@jacobgil

**Comment by jacobgil** [comment]
Hi,
Yes this needs better documentation.
I'm planning to write a tutorial for this.

- [ClassifierOutputTarget](https://github.com/jacobgil/pytorch-grad-cam/blob/master/pytorch_grad_cam/utils/model_targets.py#L6) Use the logits from a multi-category classifier. This means that the activations will correspond with whatever makes the specified category higher.
- [ClassifierOutputTarget](https://github.com/jacobgil/pytorch-grad-cam/blob/master/pytorch_grad_cam/utils/model_targets.py#L16)- Use the softmax outptus from a multi-category classifier. This means that the activations will correspond with with what makes the specified category higher, but also the other categories lower.

- [BinaryClassifierOutputTarget](https://github.com/jacobgil/pytorch-grad-cam/blob/master/pytorch_grad_cam/utils/model_targets.py#L26C7-L26C35) - For binary models, take the model output if you care about the positive category, and the negative of it if you care about the negative category.

The other targets there are for segmentation or for object detection.

Please let me know if this is clear?

**Comment by willianck** [comment]
Can the **ClassifierOutputTarget** be used if the output layer is used for regression; in this case predicting a single continuous value ?

**Comment by Longhuiberkeley** [comment]
@willianck  I am working on an image regression task. Have you figured that out?
